chore: Upgrade React Native and dependencies to latest minor versions

asdolo opened the pull request:
### Pull request type

<!-- Please check the type of change your PR introduces: -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [X] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

---

### Description

- Upgrade `react-native` from `0.72.3` to `0.72.5`.
- Upgrade all other dependencies to its latest minor version.

---

#### Notion board reference

- [Upgrade React Native and dependencies to latest minor versions](https://www.notion.so/rootstrap/Upgrade-React-Native-and-dependencies-to-latest-minor-versions-d9f30af981634f1db1c1fe6a641bef9e?pvs=4)

---

#### Tasks:

- [X] Tested on IOS
- [X] Tested on Android
- [ ] Tested on a small device
- [ ] Tested on a real device
- [ ] Tested all flows related with this PR changes
- [ ] Tested accessibility
- [ ] Added tests

marcos09 commented:
What Xcode version are you using @asdolo ? You enabled unary_functions again in the .xcodeproj but you didn't update the Podfile to enable that in your pods as it's described [here](https://github.com/facebook/react-native/issues/37748#issuecomment-1580589448)

I think we should have that flag enabled too in the pods at least the most common dependencies for RN projects are updated. WDYT?

asdolo commented:
@marcos09 I just updated the PR with the missing flags. It seems like latest version of React Native automatically handles this case (see XCode 15 fixes under [iOS specific fixes](https://github.com/facebook/react-native/blob/main/CHANGELOG.md#ios-specific)).

When I created this PR I was using Xcode 14. Now I updated to Xcode 15, and it works in both cases.
